client/rest-high-level/src/main/java/org/elasticsearch/client/Request.java

```diff
@@ -68,48 +79,207 @@ static Request get(GetRequest getRequest) {
-            if (fetchSourceContext.fetchSource() == false) {
-                params.put("_source", Boolean.FALSE.toString());
+
+        Params putParam(String key, String value) {
+            if (Strings.hasLength(value)) {
+                params.put(key, value);
             }
-            if (fetchSourceContext.includes() != null && fetchSourceContext.includes().length > 0) {
-                params.put("_source_include", String.join(",", fetchSourceContext.includes()));
+            return this;
+        }
+
+        Params putParam(String key, TimeValue value) {
+            if (value != null) {
+                return putParam(key, value.getStringRep());
             }
-            if (fetchSourceContext.excludes() != null && fetchSourceContext.excludes().length > 0) {
-                params.put("_source_exclude", String.join(",", fetchSourceContext.excludes()));
+            return this;
+        }
+
+        Params withFetchSourceContext(FetchSourceContext fetchSourceContext) {
+            if (fetchSourceContext != null) {
+                if (fetchSourceContext.fetchSource() == false) {
+                    putParam("_source", Boolean.FALSE.toString());
+                }
+                if (fetchSourceContext.includes() != null && fetchSourceContext.includes().length > 0) {
+                    putParam("_source_include", String.join(",", fetchSourceContext.includes()));
+                }
+                if (fetchSourceContext.excludes() != null && fetchSourceContext.excludes().length > 0) {
+                    putParam("_source_exclude", String.join(",", fetchSourceContext.excludes()));
+                }
             }
+            return this;
         }
-        return Collections.unmodifiableMap(params);
-    }
 
-    private static String getEndpoint(GetRequest getRequest) {
-        StringJoiner pathJoiner = new StringJoiner("/", "/", "");
-        return pathJoiner.add(getRequest.index()).add(getRequest.type()).add(getRequest.id()).toString();
-    }
+        Params withParent(String parent) {
+            return putParam("parent", parent);
+        }
+
+        Params withPipeline(String pipeline) {
+            return putParam("pipeline", pipeline);
+        }
+
+        Params withPreference(String preference) {
+            return putParam("preference", preference);
+        }
+
+        Params withRealtime(boolean realtime) {
+            if (realtime == false) {
+                return putParam("realtime", Boolean.FALSE.toString());
+            }
+            return this;
+        }
+
+        Params withRefresh(boolean refresh) {
+            if (refresh) {
+                return withRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE);
+            }
+            return this;
+        }
+
+        Params withRefreshPolicy(WriteRequest.RefreshPolicy refreshPolicy) {
+            if (refreshPolicy != WriteRequest.RefreshPolicy.NONE) {
```

Comment:
Scratch that, I see this is private and only used internally, and the request only has one of each parameters, so duplicate calls are unlikely. Just me being paranoid.